### ISSUE TITLE
IndexError: list index out of range occurs intermittently when utilizing the ECStask Block

### ISSUE THREAD

**Issue by masonmenges** [PrefectHQ/prefect-aws#151]
Intermittently when Running a flow using the ECStask block the run appears to fail with this error appearing in the agent logs `IndexError: list index out of range`. This seems to be occurring before the flow run is actually submitted so it doesn't appear tied to the flow code itself.

Agent Logs:

```
09:00:00.595 | ERROR | prefect.agent - Failed to submit flow run 'XXXXXXXXXXX' to infrastructure.

Traceback (most recent call last):

File "/usr/local/lib/python3.10/site-packages/prefect/agent.py", line 257, in _submit_run_and_capture_errors

result = await infrastructure.run(task_status=task_status)

File "/usr/local/lib/python3.10/site-packages/prefect_aws/ecs.py", line 456, in run

) = await run_sync_in_worker_thread(

File "/usr/local/lib/python3.10/site-packages/prefect/utilities/asyncutils.py", line 68, in run_sync_in_worker_thread

return await anyio.to_thread.run_sync(call, cancellable=True)

File "/usr/local/lib/python3.10/site-packages/anyio/to_thread.py", line 31, in run_sync

return await get_asynclib().run_sync_in_worker_thread(

File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread

return await future

File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 867, in run

result = context.run(func, *args)

File "/usr/local/lib/python3.10/site-packages/prefect_aws/ecs.py", line 561, in _create_task_and_wait_for_start

self._wait_for_task_start(

File "/usr/local/lib/python3.10/site-packages/prefect_aws/ecs.py", line 744, in _wait_for_task_start

for task in self._watch_task_run(

File "/usr/local/lib/python3.10/site-packages/prefect_aws/ecs.py", line 712, in _watch_task_run

task = ecs_client.describe_tasks(tasks=[task_arn], cluster=cluster_arn)[

IndexError: list index out of range
```

**Comment by gota0** [comment]
Hi @madkinsz. I'm getting the same error `IndexError: list index out of range` on prefect-aws v0.2.5. Flow runs fail before any task begins. It does not happen all the time but it happens occasionally. Is there anything I can do to prevent this? Any assistance is greatly appreciated.

agent logs: 
```
11:04:57.338 | ERROR   | prefect.agent - Failed to submit flow run 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' to infrastructure.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/prefect/agent.py", line 500, in _submit_run_and_capture_errors
    result = await infrastructure.run(task_status=task_status)
  File "/usr/local/lib/python3.8/site-packages/prefect_aws/ecs.py", line 621, in run
    ) = await run_sync_in_worker_thread(
  File "/usr/local/lib/python3.8/site-packages/prefect/utilities/asyncutils.py", line 91, in run_sync_in_worker_thread
    return await anyio.to_thread.run_sync(
  File "/usr/local/lib/python3.8/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/usr/local/lib/python3.8/site-packages/prefect_aws/ecs.py", line 814, in _create_task_and_wait_for_start
    self._report_task_run_creation_failure(task_run, exc)
  File "/usr/local/lib/python3.8/site-packages/prefect_aws/ecs.py", line 810, in _create_task_and_wait_for_start
    task = self._run_task(ecs_client, task_run)
  File "/usr/local/lib/python3.8/site-packages/prefect_aws/ecs.py", line 1487, in _run_task
    return ecs_client.run_task(**task_run)["tasks"][0]
IndexError: list index out of range
11:04:57.341 | INFO    | prefect.agent - Completed submission of flow run 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
```

**Comment by rmnvncnt** [comment]
I have the same error. @gota0 do you happen to submit a FARGATE task requiring a large ECS worker? I'm starting to think that this error occurs when the requested resource is not immediately available in your AWS region. Notably, it occurs more often to me when requesting a 8 CPU tasks than a 2 CPU one, for instance.

**Comment by THOR300** [comment]
> I have the same error. @gota0 do you happen to submit a FARGATE task requiring a large ECS worker? I'm starting to think that this error occurs when the requested resource is not immediately available in your AWS region. Notably, it occurs more often to me when requesting a 8 CPU tasks than a 2 CPU one, for instance.

@rmnvncnt I've experienced this error as well. I was attempting to run ecs tasks on a self hosted ec2 instance. I.e. connecting to the ecs cluster by starting the ecs agent. This hadn't set the correct attributes on the ec2 instance and thus the task definition could not be placed on the infrastucture resulting in the below error. 

`{
    "tasks": [],
    "failures": [
        {
            "arn": "arn:aws:ecs:eu-west-1:281621126254:container-instance/05b01a6fc2884248a6642faea40118e0",
            "reason": "ATTRIBUTE"
        }
    ]
}`

**TLDR: The correct infrastructure was not available to submit the task definition onto.**
